executor/common_linux.h

```diff
@@ -5601,8 +5601,15 @@ static void setup_swap()
 		failmsg("swap file open failed", "file: %s", SWAP_FILE);
 		return;
 	}
-	// We cannot do ftruncate -- swapon complains about this. Do fallocate instead.
-	fallocate(fd, FALLOC_FL_ZERO_RANGE, 0, SWAP_FILE_SIZE);
+	// fallocate is not available on arm32, so let's just write zeroes to the file.
```

Comment:
The EOPNOTSUPP we are getting presumably comes from here:
https://elixir.bootlin.com/linux/latest/source/fs/open.c#L321

We seem to build actual ext2 image for arm:
```
$ file buildroot-arm-2020.11
buildroot-arm-2020.11: Linux rev 1.0 ext2 filesystem data, UUID=25dc5b9c-76e5-4419-8452-235cee655fc0, volume name "rootfs" (large files)
```

And looking at our buildroot configs, amd64 has:
```
BR2_TARGET_ROOTFS_EXT2_4=y
```
while arm has:
```
BR2_TARGET_ROOTFS_EXT2_2=y
```

ext2 seems to be mostly unsupported at this point (there was a recent discussion on ext2 bug syzbot reported).

Maybe we build proper ext4 image for arm?

Reply:
I've rebuilt the image with that change, but got the same boot error @HerrSpace recently got while rebuilding the arm32 image without changes:

```
[    7.015704][    T1] Run /sbin/init as init process
[    7.382181][    T1] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000004
[    7.382823][    T1] CPU: 0 PID: 1 Comm: init Not tainted 6.4.0-rc6-syzkaller #0
[    7.383158][    T1] Hardware name: ARM-Versatile Express
[    7.383500][    T1] Backtrace: 
[    7.383802][    T1] [<817da978>] (dump_backtrace) from [<817daa6c>] (show_stack+0x18/0x1c)
[    7.384930][    T1]  r7:00000000 r6:826229c4 r5:60000093 r4:81f9dbd0
[    7.385363][    T1] [<817daa54>] (show_stack) from [<817f7554>] (dump_stack_lvl+0x48/0x54)
[    7.385840][    T1] [<817f750c>] (dump_stack_lvl) from [<817f7578>] (dump_stack+0x18/0x1c)
[    7.386248][    T1]  r5:00000000 r4:82850d14
[    7.386499][    T1] [<817f7560>] (dump_stack) from [<817db564>] (panic+0x120/0x370)
[    7.386916][    T1] [<817db444>] (panic) from [<80249448>] (do_exit+0xa98/0xa9c)
[    7.387302][    T1]  r3:00000001 r2:00000000 r1:00000004 r0:81f8ede4
[    7.387527][    T1]  r7:830f0000
[    7.387664][    T1] [<802489b0>] (do_exit) from [<80249634>] (do_group_exit+0x40/0x8c)
[    7.387962][    T1]  r7:df819f58
[    7.388100][    T1] [<802495f4>] (do_group_exit) from [<80258184>] (get_signal+0x9b4/0xa28)
[    7.388412][    T1]  r7:df819f58 r4:831e8000
[    7.388573][    T1] [<802577d0>] (get_signal) from [<8020bd14>] (do_work_pending+0x134/0x538)
[    7.388907][    T1]  r10:5ac3c35a r9:7efffd08 r8:00000000 r7:fffffe30 r6:df819f44 r5:df819fb0
[    7.389197][    T1]  r4:831e8000
[    7.389331][    T1] [<8020bbe0>] (do_work_pending) from [<80200088>] (slow_work_pending+0xc/0x20)
[    7.389703][    T1] Exception stack(0xdf819fb0 to 0xdf819ff8)
[    7.390002][    T1] 9fa0:                                     00000001 7e829df4 00000000 00000001
[    7.390320][    T1] 9fc0: 76e615dc 76f945a0 76f945a0 00000001 7e829ecc 00000000 76f9a010 00000000
[    7.390642][    T1] 9fe0: 76f990f8 7e829de0 76d97a34 76d97318 20000010 ffffffff
[    7.390935][    T1]  r10:831e8000 r9:831e8000 r8:00000000 r7:00000000 r6:80201018 r5:00000000
[    7.391220][    T1]  r4:76d97318
[    7.391909][    C1] CPU1: stopping
[    7.392860][    C1] CPU: 1 PID: 0 Comm: swapper/1 Not tainted 6.4.0-rc6-syzkaller #0
[    7.394300][    C1] Hardware name: ARM-Versatile Express
[    7.395328][    C1] Backtrace: frame pointer underflow
[    7.396599][    C1] [<817da978>] (dump_backtrace) from [<817daa6c>] (show_stack+0x18/0x1c)
[    7.398324][    C1]  r7:81f89e44 r6:828401e0 r5:600001d3 r4:81f9dbd0
[    7.399542][    C1] [<817daa54>] (show_stack) from [<817f7554>] (dump_stack_lvl+0x48/0x54)
[    7.401069][    C1] [<817f750c>] (dump_stack_lvl) from [<817f7578>] (dump_stack+0x18/0x1c)
[    7.402135][    C1]  r5:00000001 r4:00000003
[    7.402673][    C1] [<817f7560>] (dump_stack) from [<8020fde0>] (do_handle_IPI+0x29c/0x2d0)
[    7.403672][    C1] [<8020fb44>] (do_handle_IPI) from [<8020fe34>] (ipi_handler+0x20/0x28)
[    7.404691][    C1]  r9:831f4680 r8:df805f78 r7:00000014 r6:81b0febc r5:8300cb80 r4:83092a00
[    7.405680][    C1] [<8020fe14>] (ipi_handler) from [<802bf184>] (handle_percpu_devid_irq+0x9c/0x2d4)
[    7.406786][    C1] [<802bf0e8>] (handle_percpu_devid_irq) from [<802b89e8>] (generic_handle_domain_irq+0x30/0x40)
[    7.408032][    C1]  r10:00000000 r9:831f4680 r8:00000000 r7:df80a00c r6:824aebe0 r5:df80a000
[    7.409010][    C1]  r4:8260cdf0 r3:00010001
[    7.409534][    C1] [<802b89b8>] (generic_handle_domain_irq) from [<8087d78c>] (gic_handle_irq+0x68/0x7c)
[    7.410672][    C1] [<8087d724>] (gic_handle_irq) from [<817f7c30>] (generic_handle_arch_irq+0x60/0x80)
[    7.411731][    C1]  r7:df85df10 r6:8211b910 r5:82156128 r4:824b0264
[    7.412231][    C1] [<817f7bd0>] (generic_handle_arch_irq) from [<817acba0>] (call_with_stack+0x1c/0x20)
[    7.413015][    C1]  r9:831f4680 r8:00000000 r7:df85df44 r6:ffffffff r5:20000113 r4:817f9250
[    7.413652][    C1] [<817acb84>] (call_with_stack) from [<80200b44>] (__irq_svc+0x84/0xac)
[    7.414323][    C1] Exception stack(0xdf85df10 to 0xdf85df58)
[    7.414806][    C1] df00:                                     00000001 81f9dbd0 00024c1c 00000001
[    7.415524][    C1] df20: 831f4680 8260c498 00000001 8260c4e0 00000000 827e16e6 00000000 df85df7c
[    7.416216][    C1] df40: df85df50 df85df60 817f87e4 817f9250 20000113 ffffffff
[    7.416784][    C1] [<817f9204>] (default_idle_call) from [<80293834>] (do_idle+0x280/0x2f0)
[    7.417492][    C1]  r7:8260c4e0 r6:831f4680 r5:8260c498 r4:00000001
[    7.417985][    C1] [<802935b4>] (do_idle) from [<80293bc0>] (cpu_startup_entry+0x20/0x24)
[    7.418693][    C1]  r10:00000000 r9:412fc0f1 r8:80003010 r7:82850464 r6:831f4680 r5:00000001
[    7.419358][    C1]  r4:00000097
[    7.419632][    C1] [<80293ba0>] (cpu_startup_entry) from [<80210394>] (secondary_start_kernel+0x12c/0x188)
[    7.420410][    C1] [<80210268>] (secondary_start_kernel) from [<802016f4>] (__enable_mmu+0x0/0xc)
```

Reply:
OK, let's just write to the file.

Reply:
Then update the comment for future generations. arm32 support fallocate on its own.

Reply:
Isn't all arm32 hardware like 10+ years old at this point? How much legacy arm32s that runs linux and gets kernel updates is really out there?

Maybe we can drop this architecture.